library/std/src/thread/local.rs

```diff
@@ -134,10 +134,24 @@ impl<T: 'static> fmt::Debug for LocalKey<T> {
 /// thread_local! {
 ///     pub static FOO: RefCell<u32> = RefCell::new(1);
 ///
-///     #[allow(unused)]
 ///     static BAR: RefCell<f32> = RefCell::new(1.0);
 /// }
-/// # fn main() {}
+///
+/// FOO.with(|foo| assert_eq!(*foo.borrow(), 1));
+/// BAR.with(|bar| assert_eq!(*bar.borrow(), 1.0));
+/// ```
+///
+/// It also allows wrapping the initializer in `const {}` (custom syntax for this macro), which
+/// allows it to use a more efficient implementation.
+/// For this, the initialized has to be evaluatable in a `const` context.
```

Comment:
```suggestion
/// This macro supports a special `const {}` syntax that can be used
/// when the initialization expression can be evaluated as a constant.
/// This enables a more efficient thread local implementation that
/// can avoid lazy initialization. For types that do not
/// [need to be dropped][std::mem::needs_drop], this enables an
/// even more efficient implementation that does not need to
/// track any additional state.
```

Reply:
The implementation is only available on platforms with native TLS. I think "this can enable" would be more accurate.